docs/Doxyfile.config

```diff
@@ -823,8 +825,27 @@ WARN_LOGFILE           = "@DOCS_BUILD_DIR@/ie_docs.log"
 # spaces. See also FILE_PATTERNS and EXTENSION_MAPPING
 # Note: If this tag is empty the current directory is searched.
 
-INPUT                  = "@DOCS_BUILD_DIR@" \
-                         "@OV_INFERENCE_DIR@/include"
+INPUT                  = "@MARKDOWN_INPUT@" \
+                         "@OpenVINO_SOURCE_DIR@/src/frontends/common/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/conditional_compilation/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/itt/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/legacy/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/legacy/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/low_precision_transformations/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/low_precision_transformations/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/offline_transformations/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/preprocessing/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/snippets/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/transformations/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/common/util/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/core/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/core/builder/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/core/reference/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/core/shape_inference/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/frontends/common/include" \
+                         "@OpenVINO_SOURCE_DIR@/src/inference/dev_api" \
+                         "@OpenVINO_SOURCE_DIR@/src/inference/include"
```

Comment:
from these header files we need only public API ones:
 "@OpenVINO_SOURCE_DIR@/src/frontends/common/include" \
                         "@OpenVINO_SOURCE_DIR@/src/core/include" \
                         "@OpenVINO_SOURCE_DIR@/src/inference/include"
                         "@OpenVINO_SOURCE_DIR@/src/frontends/common/include" \